src/Hydrator.php

```diff
@@ -42,194 +33,128 @@ final class Hydrator implements HydratorInterface
      * @var ParameterAttributesHandler Parameter attributes handler.
      */
     private ParameterAttributesHandler $parameterAttributesHandler;
+    private ObjectPropertiesExtractor $objectPropertiesExtractor;
+    private DataPropertyAccessor $dataPropertyAccessor;
 
     /**
-     * @param ContainerInterface $container Container used to resolve created object dependencies and get attributes'
-     * resolvers.
      * @param TypeCasterInterface|null $typeCaster Type caster used to cast raw values.
      */
-    public function __construct(ContainerInterface $container, ?TypeCasterInterface $typeCaster = null)
-    {
-        $this->injector = new Injector($container);
+    public function __construct(
+        ?TypeCasterInterface $typeCaster = null,
+        ?AttributeResolverInitiator $initiator = null,
```

Comment:
Maybe create `AttributeResolverInitiatorInterface` and create two implementation in package:

- create via container,
- create resolvers without constructor arguments only.

And interface will allow users to create own implementations.